### PR TITLE
Passing PDS Test Params as JSON file

### PR DESCRIPTION
**Pull request by Madanagopal19**
Passing PDS Test Params as JSON file,
This will help in reducing env params and in test runs

Introduced a new flag `--pds-parameter-json` if this flag is passed with the json file params it will honour those params else it uses the default json file
Example Json file
`{
  "DataServiceToTest": [
    {
      "Name": "RabbitMQ",
      "Version": "3.10.9",
      "Image": "716b7b8",
      "Replicas": 1,
      "ScaleReplicas": 6,
      "OldVersion": "3.10.7",
      "OldImage": "862425e"
    },
    {
      "Name": "MySQL",
      "Version": "8.0.30",
      "Image": "a7bc638",
      "Replicas": 2,
      "ScaleReplicas": 6,
      "OldVersion": "8.0.30",
      "OldImage": "a7bc638"
    },
    {
      "Name": "PostgreSQL",
      "Version": "14.5",
      "Image": "d69b376",
      "Replicas": 3,
      "ScaleReplicas": 6,
      "OldVersion": "14.4",
      "OldImage": "b64d741"
    }
  ],
  "InfraToTest": {
    "ControlPlaneURL": "https://staging.pds.portworx.com/",
    "AccountName": "Portworx",
    "ClusterType": "onprem",
    "Namespace": "automation"
  }
}`

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -154,6 +154,8 @@ const (
 	configMapName                        = "kubeconfigs"
 	pxNamespace                          = "kube-system"
 
+	pdsParamCliFlag = "pdsparameter"
```
Comment (lsrinivas-pure) [inline tests/common.go:157]: chnage it to pds-parameter-json

File: tests/common.go [hunk 2]
```diff
@@ -218,7 +220,7 @@ const (
 	defaultNodeDriver                     = "ssh"
 	defaultMonitorDriver                  = "prometheus"
 	defaultStorageDriver                  = "pxd"
-	defaultLogLocation                    = "/testresults/"
+	defaultLogLocation                    = "/Users/marunachalam/Desktop"
```
Comment (lsrinivas-pure) [inline tests/common.go:223]: revert this change

File: tests/pds/dataservice_test.go [hunk 3]
```diff
@@ -82,54 +66,30 @@ func TestDataService(t *testing.T) {
...
-		TargetClusterName := pdslib.GetAndExpectStringEnvVar(envTargetClusterName)
-		Expect(TargetClusterName).NotTo(BeEmpty(), "ENV "+envTargetClusterName+" is not set")
-
 		DeployAllDataService, err = pdslib.GetAndExpectBoolEnvVar(envDeployAllDataService)
 		Expect(err).NotTo(HaveOccurred())
 
-		DeployAllVersions, err = pdslib.GetAndExpectBoolEnvVar(envDeployAllVersions)
+		pdsparams := Inst().PDSParams
+		params, err = pdslib.ReadParams(pdsparams)
 		Expect(err).NotTo(HaveOccurred())
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:74]: change this to dash verifications

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: deployments/deploy-ssh.sh [hunk 4]
```diff
@@ -464,6 +464,7 @@ spec:
             "--testset-id=$TESTSET_ID",
             "--branch=$BRANCH",
             "--product=$PRODUCT",
+            "--pdsparameter=$PDS_PARAMETER_JSON",
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply deployments/deploy-ssh.sh]: done

File: drivers/pds/lib/utils.go [hunk 5]
```diff
@@ -222,11 +250,16 @@ func SetupPDSTest(ControlPlaneURL, ClusterType, TargetClusterName, AccountName s
 	projectName := projects[0].GetName()
 	logrus.Infof("Project Details- Name: %s, UUID: %s ", projectName, projectID)
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go:251]: In next automation sprint we will adapt to dash. In this manual sprint we are targeting only the parameter part to run tests from Jenkins

File: drivers/pds/lib/utils.go [hunk 6]
```diff
@@ -222,11 +250,16 @@ func SetupPDSTest(ControlPlaneURL, ClusterType, TargetClusterName, AccountName s
...
+	if err != nil {
+		logrus.Errorf("Error while getting k8s namespace %v", err)
+		return "", "", "", "", "", err
+	}
+	clusterID := string(ns.GetObjectMeta().GetUID())
 	if len(clusterID) > 0 {
 		logrus.Infof("clusterID %v", clusterID)
 	} else {
-		logrus.Errorf("Cluster ID is empty %v", clusterID)
+		logrus.Errorf("Cluster ID is empty")
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go:262]: In next automation sprint we will adapt to dash. In this manual sprint we are targeting only the parameter part to run tests from Jenkins

File: tests/common.go (hunk 1 above)
Comment (Madanagopal19) [reply tests/common.go]: done

File: tests/common.go (hunk 2 above)
Comment (Madanagopal19) [reply tests/common.go]: done

File: tests/pds/dataservice_test.go (hunk 3 above)
Comment (Madanagopal19) [reply tests/pds/dataservice_test.go:74]: In next automation sprint we will adapt to dash. In this manual sprint we are targeting only the parameter part to run tests from Jenkins

File: tests/common.go [hunk 7]
```diff
@@ -154,6 +154,8 @@ const (
 	configMapName                        = "kubeconfigs"
 	pxNamespace                          = "kube-system"
 
+	pdsParamCliFlag = "pds-parameter-json"
+
```
Comment (lsrinivas-pure) [inline tests/common.go:158]: format as line same as other lines

File: tests/common.go [hunk 8]
```diff
@@ -3798,6 +3802,11 @@ func ParseFlags() {
 		log.Fatalf("failed to parse app list: %v. err: %v", appListCSV, err)
 	}
 
+	// pdsparameters, err := ioutil.ReadFile(pdsParam)
+	// if err != nil {
+	// 	log.Fatalf("File error: %v\n", err)
```
Comment (lsrinivas-pure) [inline tests/common.go:3807]: remove this if not required

File: tests/common.go [hunk 9]
```diff
@@ -3959,17 +3969,17 @@ func setLoglevel(tpLog *logrus.Logger, logLevel string) {
...
 	tpLog.Infof("Log Dir: %s", f.Name())
 }
 
-//CreateLogFile creates file and return the file object
+// CreateLogFile creates file and return the file object
 func CreateLogFile(filename string) *os.File {
 	var filePath string
 	if strings.Contains(filename, "/") {
-		filePath = filename
+		filePath = "/Users/marunachalam/Desktop/torpedo.log"
```
Comment (lsrinivas-pure) [inline tests/common.go:3982]: revert this hard coded value

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/common.go (hunk 7 above)
Comment (Madanagopal19) [reply tests/common.go:157]: done

File: tests/common.go (hunk 8 above)
Comment (Madanagopal19) [reply tests/common.go]: done

File: tests/common.go (hunk 9 above)
Comment (Madanagopal19) [reply tests/common.go]: done

File: drivers/pds/lib/utils.go [hunk 10]
```diff
@@ -322,8 +446,36 @@ func GetResourceTemplate(tenantID string, supportedDataServices []string) (map[s
...
+		return "", err
+	}
+	isavailable = false
+	isTemplateavailable = false
+	dataServiceId := GetDataServiceID(supportedDataService)
+
+	for i := 0; i < len(appConfigs); i++ {
+		if appConfigs[i].GetName() == appConfigTemplateName {
+			isTemplateavailable = true
+			//for key := range dataServiceNameIDMap {
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:462]: remove commented lines if not required

File: drivers/pds/lib/utils.go [hunk 11]
```diff
@@ -322,8 +446,36 @@ func GetResourceTemplate(tenantID string, supportedDataServices []string) (map[s
...
+			if dataServiceId == appConfigs[i].GetDataServiceId() {
+				//dataServiceNameDefaultAppConfigMap[key] = appConfigs[i].GetId()
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:464]: same here

File: drivers/pds/lib/utils.go [hunk 12]
```diff
@@ -940,25 +1163,78 @@ func GetAllSupportedDataServices() map[string]string {
...
-				return nil, err
+func UpdateDataServices(deploymentID string, appConfigID string, imageID string, nodeCount int32, resourceTemplateID, namespace string) (*pds.ModelsDeployment, error) {
+
+	logrus.Infof("depID %v appConfID %v imageID %v nodeCount %v resourceTemplateID %v", deploymentID, appConfigID, imageID, nodeCount, resourceTemplateID)
+	deployment, err = components.DataServiceDeployment.UpdateDeployment(deploymentID, appConfigID, imageID, nodeCount, resourceTemplateID, nil)
+	if err != nil {
+		logrus.Errorf("An Error Occured while updating deployment %v", err)
+		return nil, err
+	}
+	ValidateDataServiceDeployment(deployment, namespace)
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:1174]: handle the error here

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: drivers/pds/lib/utils.go (hunk 10 above)
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go]: done

File: drivers/pds/lib/utils.go (hunk 11 above)
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go]: done

File: drivers/pds/lib/utils.go (hunk 12 above)
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go]: done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM
